homeassistant/components/yolink/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""The yolink integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+from yolink.client import YoLinkClient
+from yolink.const import OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+from yolink.mqtt_client import MqttClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import (
+    aiohttp_client,
+    config_entry_oauth2_flow,
+    config_validation as cv,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from . import api, config_flow
+from .const import ATTR_CLIENT, ATTR_COORDINATOR, ATTR_MQTT_CLIENT, DOMAIN
+from .coordinator import YoLinkCoordinator
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_CLIENT_ID): cv.string,
+                vol.Required(CONF_CLIENT_SECRET): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
Should we start using <https://developers.home-assistant.io/docs/core/platform/application_credentials> ?

As in, this would introduce something we would try to get rid of?

Reply:
Can i do this later? I want to finish integration first, Just add a Sensor platform for battery entity currently.

Reply:
Yes/no. That is the thing I'm trying to raise: Adding/merging the current state now, would mean that we merge something that will have a known migration, deprecation and breaking change up ahead in the near future.

Reply:
Thanks, I need storage the client_id and secret to somewhere Nabu Casa

Reply:
We can import the YAML to application credentials, but there's no frontend support merged yet for application credentials, so we still need the YAML section here. As frontend support for application credentials isn't ready yet, I don't think we need to use application credentials in this PR but can follow up with that asap when frontend support is ready, hopefully before the beta cut in two weeks.

https://github.com/home-assistant/frontend/pull/12344

Reply:
Aah, right 👍 

I guess that makes sense, ok 👍